File: Find_Target/find.py
from typing import List, Tuple


class Solution:
    def targetIndices(self, nums: List[int], target: int) -> List[int]:
        nums = sorted(nums)

        indices: List[int] = []
        low, high = find_bound(nums, target)

        index = low

        while index <= high:
            if nums[index] == target:
                indices.append(index)

            index += 1

        return indices


def find_bound(nums: List[int], target: int) -> Tuple[int, int]:
    i, j = 0, len(nums) - 1

    while i <= j:
        mid = (i + j) // 2

        if nums[mid] > target:
            j = mid - 1
        elif nums[mid] < target:
            i = mid + 1
        else:
            break

    return (i, j)

File: Find_Target/find.cs
using System;
using System.Collections.Generic;

public class Solution
{
    public IList<int> TargetIndices(int[] nums, int target)
    {
        List<int> result = new List<int>();
        Array.Sort(nums);

        int i = findLowerBound(nums, target);
        int j = findUpperBoud(nums, target);

        for (int k = i; k <= j; k++)
        {
            if (nums[k] == target)
            {
                result.Add(k);
            }
        }

        return result;
    }

    private int findLowerBound(int[] nums, int target)
    {
        int i = 0,
            j = nums.Length - 1;

        while (i <= j)
        {
            int mid = (i + j) / 2;

            if (nums[mid] > target)
            {
                j = mid - 1;
            }
            else if (nums[mid] < target)
            {
                i = mid + 1;
            }
            else
            {
                break;
            }
        }
        return i;
    }

    private int findUpperBoud(int[] nums, int target)
    {
        int i = 0,
            j = nums.Length - 1;

        while (i <= j)
        {
            int mid = (i + j) / 2;
            if (nums[mid] > target)
            {
                j = mid - 1;
            }
            else if (nums[mid] < target)
            {
                i = mid + 1;
            }
            else
            {
                break;
            }
        }

        return j;
    }
}
